modules/context-based-restrictions/src/test/java/com/ibm/cloud/platform_services/context_based_restrictions/v1/ContextBasedRestrictionsIT.java

```diff
@@ -0,0 +1,805 @@
+            // Validate response
+            assertNotNull(response);
+            assertEquals(response.getStatusCode(), 200);
+
+            OutZonePage outZonePageResult = response.getResult();
+
+            assertNotNull(outZonePageResult);
+        } catch (ServiceResponseException e) {
+            fail(String.format("Service returned status code %d: %s%nError details: %s",
+                    e.getStatusCode(), e.getMessage(), e.getDebuggingInfo()));
+        }
+    }
+
+
+    @Test(groups = "listZones", dependsOnMethods = "testListZones")
+    public void testListZonesInvalidAccountID() throws Exception {
+        try {
+            ListZonesOptions listZonesOptions = new ListZonesOptions.Builder()
+                    .accountId(InvalidID)
+                    .transactionId(getTransactionID())
+                    .build();
+
+            // Invoke operation
+            Response<OutZonePage> response = service.listZones(listZonesOptions).execute();
+        } catch (ServiceResponseException e) {
+            if (e.getStatusCode() != 400)
+                fail(String.format("Service returned status code %d: %s%nError details: %s",
+                        e.getStatusCode(), e.getMessage(), e.getDebuggingInfo()));
+        }
+    }
+
+    @Test(groups = "getZone", dependsOnGroups = "listZones")
+    public void testGetZone() throws Exception {
+        try {
+            GetZoneOptions getZoneOptions = new GetZoneOptions.Builder()
+                    .zoneId(zoneID)
+                    .transactionId(getTransactionID())
+                    .build();
+
+            // Invoke operation
+            Response<OutZone> response = service.getZone(getZoneOptions).execute();
+            // Validate response
+            assertNotNull(response);
+            assertEquals(response.getStatusCode(), 200);
+
+            OutZone outZoneResult = response.getResult();
+
+            assertNotNull(outZoneResult);
+        } catch (ServiceResponseException e) {
+            fail(String.format("Service returned status code %d: %s%nError details: %s",
+                    e.getStatusCode(), e.getMessage(), e.getDebuggingInfo()));
+        }
+    }
+
+    @Test(groups = "getZone", dependsOnMethods = "testGetZone")
+    public void testGetZoneNotFound() throws Exception {
+        try {
+            GetZoneOptions getZoneOptions = new GetZoneOptions.Builder()
+                    .zoneId(NonExistentID)
+                    .transactionId(getTransactionID())
+                    .build();
+
+            // Invoke operation
+            Response<OutZone> response = service.getZone(getZoneOptions).execute();
+        } catch (ServiceResponseException e) {
+            if (e.getStatusCode() != 404)
+                fail(String.format("Service returned status code %d: %s%nError details: %s",
+                        e.getStatusCode(), e.getMessage(), e.getDebuggingInfo()));
+        }
+    }
```

Comment:
I'd like to ask a test which exercises status code `400`. Go and Node sdks can be a good example.

Reply:
The Go and Node examples use a Missing Parameter error to catch this, unfortunately this does not work in java as it fails in the builder, before we can even make the request